### PR TITLE
Migrate requests to httpx

### LINKED ISSUE #282: Migrate `requests` dependency to `httpx`
mostly API compatible, drop-in replacement with support for AsyncIO

### PR DESCRIPTION
**Pull request by irux**
closes #282

### CODE REVIEW

File: kpops/component_handlers/utils/exception.py [hunk 1]
```diff
@@ -1,20 +1,20 @@
 import logging
 
-import requests
+import httpx
 
 log = logging.getLogger("RequestException")
 
 
 class RequestsException(Exception):
```
Comment (disrupted) [inline kpops/component_handlers/utils/exception.py:8]: rename

File: kpops/component_handlers/topic/proxy_wrapper.py [hunk 2]
```diff
@@ -44,8 +44,8 @@ def cluster_id(self) -> str:
         bootstrap.servers configuration. Therefore, only one Kafka cluster will be returned.
         :return: The Kafka cluster ID.
         """
-        response = requests.get(url=f"{self._host}/v3/clusters")
-        if response.status_code == requests.status_codes.codes.ok:
+        response = httpx.get(url=f"{self._host}/v3/clusters")
```
Comment (disrupted) [inline kpops/component_handlers/topic/proxy_wrapper.py:47]: we should consider creating a httpx Client and setting `self._host` as base url

File: pyproject.toml
Comment (disrupted) [inline pyproject.toml:1]: remove `requests` and `responses`

File: pyproject.toml [hunk 3]
```diff
@@ -63,6 +64,10 @@ mkdocs-material-extensions = "^1.1.1"
 mkdocs-glightbox = "^0.3.1"
 mike = "^1.1.2"
 
+
+[tool.poetry.group.dependencies.dependencies]
+httpx = "^0.24.1"
+
```
Comment (disrupted) [inline pyproject.toml:70]: ?
should be under main dependencies

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py [hunk 4]
```diff
@@ -76,8 +75,9 @@ def test_should_create_post_requests_for_given_connector_configuration(
             },
         )
 
-    @responses.activate
-    def test_should_return_correct_response_when_connector_created(self):
+    def test_should_return_correct_response_when_connector_created(
+        self, httpx_mock: HTTPXMock
```
Comment (disrupted) [inline tests/component_handlers/kafka_connect/test_connect_wrapper.py:79]: nice! 🤩

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py [hunk 5]
```diff
@@ -48,7 +47,7 @@ def test_should_through_exception_when_host_is_not_set(self):
             == "The Kafka Connect host is not set. Please set the host in the config."
         )
 
-    @patch("requests.post")
+    @patch("httpx.post")
```
Comment (disrupted) [inline tests/component_handlers/kafka_connect/test_connect_wrapper.py:50]: can this be replaced with HTTPXMock?

**Review by disrupted — CHANGES_REQUESTED** [review]

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py (hunk 5 above)
Comment (irux) [reply tests/component_handlers/kafka_connect/test_connect_wrapper.py:50]: httpx mock is for the responses but not for call assertions. For this we use the patch.

File: pyproject.toml (hunk 3 above)
Comment (disrupted) [reply pyproject.toml:70]: it shouldn't be in a separate group

File: kpops/component_handlers/topic/proxy_wrapper.py (hunk 2 above)
Comment (irux) [reply kpops/component_handlers/topic/proxy_wrapper.py:47]: I would do this for the async part, because the client is going to be async too.

Comment (disrupted) [reply kpops/component_handlers/topic/proxy_wrapper.py:47]: ok, sounds good

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py (hunk 5 above)
Comment (raminqaf) [inline tests/component_handlers/kafka_connect/test_connect_wrapper.py:50]: How about we use `@pytest.fixture` for this?

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py [hunk 6]
```diff
@@ -76,8 +75,9 @@ def test_should_create_post_requests_for_given_connector_configuration(
             },
         )
 
-    @responses.activate
-    def test_should_return_correct_response_when_connector_created(self):
+    def test_should_return_correct_response_when_connector_created(
```
[the remark this answers is not in the document]
Comment (irux) [reply tests/component_handlers/kafka_connect/test_connect_wrapper.py:78]: https://colin-b.github.io/pytest_httpx/

File: kpops/component_handlers/utils/exception.py [hunk 7]
```diff
@@ -1,20 +1,20 @@
 import logging
 
-import requests
+import httpx
 
 log = logging.getLogger("RequestException")
```
Comment (disrupted) [inline kpops/component_handlers/utils/exception.py:5]: this also needs to be renamed to match the class name

Comment (irux) [reply kpops/component_handlers/utils/exception.py:5]: good point!

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (raminqaf) [reply tests/component_handlers/kafka_connect/test_connect_wrapper.py:78]: ah okay all good then

File: pyproject.toml [hunk 8]
```diff
@@ -31,7 +31,6 @@ python = "^3.10"
 pydantic = { extras = ["dotenv"], version = "^1.10.8" }
 rich = "^12.4.4"
 PyYAML = "^6.0"
-requests = "^2.28.0"
```
Comment (disrupted) [inline pyproject.toml:34]: lockfile needs to be updated as well. `poetry lock --no-update`

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py (hunk 5 above)
Comment (irux) [reply tests/component_handlers/kafka_connect/test_connect_wrapper.py:50]: the thing is you would need to know what operation to patch. You need to say if you want a post or get (etc) request. I would need to write it manually if I do like that and declare the async mock manually

**Review by disrupted — APPROVED** [review]

File: tests/component_handlers/kafka_connect/test_connect_wrapper.py (hunk 5 above)
Comment (raminqaf) [reply tests/component_handlers/kafka_connect/test_connect_wrapper.py:50]: This is similar to the logs (like log_info, log_debug, etc.), but it is not essential to change. I just wanted to keep it consistent like the other tests.

**Review by raminqaf — APPROVED** [review]

**Pull request merged**